enterprise/utils.py

```diff
@@ -512,3 +512,17 @@ def safe_extract_key(data, key, default=''):
         return data[key]
     else:
         return default
+
+
+def filter_audit_course_modes(enterprise_customer, course_modes):
+    """
+    Filter audit course modes out if the enterprise customer has the 'Hide audit track' flag enabled.
+
+    Arguments:
+        enterprise_customer: The EnterpriseCustomer that the enrollment was created using.
+        course_modes: iterable with dictionaries containing a required 'mode' key
+    """
+    if not enterprise_customer.hide_audit_track:
+        return course_modes
+    audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit'])
```

Comment:
We'll need to add this setting to [common.py](https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L3126) and [aws.py](https://github.com/edx/edx-platform/blob/master/lms/envs/aws.py#L910)